platforms/nuttx/src/canbootloader/uavcan/main.c

```diff
@@ -1184,6 +1184,13 @@ __EXPORT int main(int argc, char *argv[])
 
 
 	} else {
+		// If app is already valid, jump straight to app
+		if ((bootloader.app_valid) && (bootloader.fw_image_descriptor->major_version > 0)) {
+			common.node_id = 0; // For normal boots there is nothing connected, we need to do dynamic ID allocation
+			goto boot;
+		}
+
+		// Otherwise this is a bad/blank app and we need to be updated
```

Comment:
> @davids5 Daniel and I are revisiting this PR and would like to get it in. ...  Even erasing a sector of the app firmware and verifying the invalid app doesn't boot and gets re-flashed.

@AlexKlimaj - that is not the concern. Please add 'hardfault_log fault 1' (or the equivalent  ASSERT(1)) to the start up and ensure the device can be updated. Test it with this change and without it.

Reply:
Okay I think I have it sorted.